virttest/qemu_vm.py

```diff
@@ -2912,29 +2912,21 @@ def create(self, name=None, params=None, root_dir=None,
                 self.monitors.append(monitor)
 
             # Create virtio_ports (virtio_serialports and virtio_consoles)
-            i = 0
-            self.virtio_ports = []
-            for port in params.objects("virtio_ports"):
-                port_params = params.object_params(port)
-                if port_params.get('virtio_port_chardev') == "spicevmc":
-                    filename = 'dev%s' % port
-                else:
-                    filename = self.get_virtio_port_filename(port)
-                port_name = port_params.get('virtio_port_name_prefix', None)
-                if port_name:   # If port_name_prefix was used
-                    port_name = port_name + str(i)
-                else:           # Implicit name - port
-                    port_name = port
-                if port_params.get('virtio_port_type') in ("console",
-                                                           "virtio_console"):
-                    self.virtio_ports.append(
-                        qemu_virtio_port.VirtioConsole(port, port_name,
-                                                       filename))
-                else:
-                    self.virtio_ports.append(
-                        qemu_virtio_port.VirtioSerial(port, port_name,
-                                                      filename))
-                i += 1
+            try:
+                for name in params.get('virtio_ports').split():
```

Comment:
For tests which don't have virtio_ports parameter set, this line of code will throw "'NoneType' object has no attribute 'split'" error. Should change it to:
<pre><code> for name in params.objects("virtio_ports"):</code></pre>